go/pkg/vpnkit/forward/vmnet_darwin.go

```diff
@@ -142,16 +142,38 @@ func closeUDPVmnet(IP net.IP, Port uint16, l libproxy.UDPListener) error {
 }
 
 func switchFDs(raw syscall.RawConn, newFD uintptr) error {
-	var dupErr error
+	var controlErr error
 	err := raw.Control(func(fd uintptr) {
-		dupErr = syscall.Dup2(int(newFD), int(fd))
+		controlErr = syscall.Dup2(int(newFD), int(fd))
+		if controlErr != nil {
+			return
+		}
+		// Go normally opens file descriptors with O_CLOEXEC. This prevents races where
```

Comment:
I doubt `syscall.Dup2` would set the `O_CLOEXEC` on the new fd. Do you have a reference for this comment?
Have you considered `syscall.Dup3` ? Though I have no idea how it behaves on mac.

Reply:
Ah sorry I should have squashed this commit -- the code is deleted in a later patch. I had a look on Mac for `Dup3` but unfortunately they don't have it :(